src/iosrv/srv.c

```diff
@@ -664,11 +747,31 @@ static int
 dss_start_one_xstream(hwloc_cpuset_t cpus, int xs_id)
 {
 	struct dss_xstream	*dx;
+	hwloc_obj_t		pu;
 	ABT_thread_attr		attr = ABT_THREAD_ATTR_NULL;
 	int			rc = 0;
 	bool			comm; /* true to create cart ctx for RPC */
 	int			xs_offset = 0;
 
+	/** Retrieve first processing unit (i.e. hyperthread) */
+	pu = hwloc_get_obj_inside_cpuset_by_type(dss_topo, cpus, HWLOC_OBJ_PU,
+						 0);
+	if (pu == NULL) {
+		D_ERROR("No hyperthread found!\n");
+	} else {
+		hwloc_cpuset_t puset;
+
+		/** overwrite cpuset to bind xstream to the first hyperthread */
+		puset = pu->cpuset;
+		/** check whether a second hyperthread is available (SMT on) */
+		pu = hwloc_get_obj_inside_cpuset_by_type(dss_topo, cpus,
+							 HWLOC_OBJ_PU, 1);
+		if (pu == NULL)
+			D_ERROR("SMT is disabled\n");
```

Comment:
Should it be an error message or just debug message? Is SMT always enabled on intel CPU by default?